tests/v3_api/test_secrets.py

```diff
@@ -0,0 +1,331 @@
+        key = k
+        vdecoded = base64.b64decode(v)
+
+    assert key == keyvaluepair.keys()[0]
+    assert vdecoded == keyvaluepair.values()[0]
+
+    newkeyvaluepair = {"testallnew": "valueallnew"}
+    updated_dict = {"testall": "valueall", "testallnew": "valueallnew"}
+    updated_secret = p_client.update(secret, name=name, namespaceId='NULL',
+                                     stringData=newkeyvaluepair)
+
+    assert updated_secret['baseType'] == "secret"
+    updatedsecretdata = updated_secret['data']
+
+    print updatedsecretdata
+
+    finalkeyvaluepair = {}
+    for k, v in updatedsecretdata.items():
+        keytest = k
+        vdecoded = base64.b64decode(v)
+        finalkeyvaluepair.update({keytest: vdecoded})
+
+    assert len(updatedsecretdata.items()) == 2
+    print finalkeyvaluepair
+    assert cmp(updated_dict, finalkeyvaluepair) == 0
+
+
+def test_edit_secret_single_ns():
+
+    p_client = namespace["p_client"]
+    ns = namespace["ns"]
+    name = random_test_name("default")
+    keyvaluepair = {"testall": "valueall"}
+
+    secret = p_client.create_namespaced_secret(name=name,
+                                               namespaceId=ns['name'],
+                                               stringData=keyvaluepair)
+    print secret
+    assert secret['baseType'] == "namespacedSecret"
+    secretdata = secret['data']
+
+    for k, v in secretdata.items():
+        key = k
+        vdecoded = base64.b64decode(v)
+
+    assert key == keyvaluepair.keys()[0]
+    assert vdecoded == keyvaluepair.values()[0]
+
+    newkeyvaluepair = {"testallnew": "valueallnew"}
+    updated_dict = {"testall": "valueall", "testallnew": "valueallnew"}
+    updated_secret = p_client.update(secret, name=name,
+                                     namespaceId=ns['name'],
+                                     stringData=newkeyvaluepair)
+
+    print secret
+    assert updated_secret['baseType'] == "namespacedSecret"
+    updatedsecretdata = updated_secret['data']
+
+    print updatedsecretdata
+
+    finalkeyvaluepair = {}
+    for k, v in updatedsecretdata.items():
+        keytest = k
+        vdecoded = base64.b64decode(v)
+        finalkeyvaluepair.update({keytest: vdecoded})
+
+    assert len(updatedsecretdata.items()) == 2
+    print finalkeyvaluepair
+    assert cmp(updated_dict, finalkeyvaluepair) == 0
+
+
+@pytest.fixture(scope='module', autouse="True")
+def create_project_client(request):
+
+    client = get_admin_client()
+    clusters = client.list_cluster()
+    assert len(clusters) >= 1
+    cluster = clusters[0]
+    create_kubeconfig(cluster)
+    p, ns = create_project_and_ns(ADMIN_TOKEN, cluster)
+    p_client = get_project_client_for_token(p, ADMIN_TOKEN)
+    namespace["p_client"] = p_client
+    namespace["ns"] = ns
+    namespace["cluster"] = cluster
+    namespace["project"] = p
+
+    def fin():
+        client = get_admin_client()
+        client.delete(namespace["project"])
+    request.addfinalizer(fin)
+
+
```

Comment:
can this method be changed to take in secret object instead of keyvaluepair?

Reply:
The method can handle multiple keyvalue pairs